src/Microsoft.IdentityModel.Tokens/Exceptions/SecurityTokenExpiredException.cs

```diff
@@ -73,6 +73,15 @@ public SecurityTokenExpiredException(string message, Exception inner)
         protected SecurityTokenExpiredException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
+            Expires = (DateTime)info.GetValue(nameof(Expires), typeof(DateTime));
+        }
+
+        /// <inheritdoc/>
+        public override void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            base.GetObjectData(info, context);
+
+            info.AddValue(nameof(Expires), Expires, typeof(DateTime));
```

Comment:
Since this could be derived from, a users could set the same value.
If we use the fully qualified name such as: GetType().FullName + "." + nameof(Expires), the collusions would be reduced.
Also this can throw, if the type cast fails on info.GetValue.

Have a look at: https://referencesource.microsoft.com/#mscorlib/system/security/claims/ClaimsIdentity.cs,8437a543724aff81 where fully qualified names are used. Also the SerializationInfo enumerator is queried for values. See the Deserialize method.

Reply:
Thanks for the link! I will refer to it.

How should we handle throwing an exception if the type cast fails on GetValue? We can check the type based on the enumerator, but silently failing doesn't seem right here either.

Reply:
@brentschmaltz @GeoK perspectives?

Reply:
IMO, we should not do anything, we should let the original exception be thrown as described here.
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.serializationinfo.getvalue?view=netcore-3.1

@keegan-caruso  Can you elaborate on when/how this situation will occur?

Reply:
@henrik-me should only happen in error / not during normal execution. If a user derived from the class and mistakenly created a conflict with a different type for example.

Reply:
we should add documentation i opened: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1408